src/cycax/cycad/cycad_side.py

```diff
@@ -0,0 +1,463 @@
+#!/usr/bin/env python3
+
+
+class CycaxSide:
+    def __init__(self, parent):
+        self._parent = parent
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return "Side: " + self.name
+
+    def hole(
+        self,
+        pos: tuple[float, float],
+        diameter: float,
+        depth: float = None,
+        inner: bool = True,
+        external_only: bool = False,
+    ):
+        """
+        This will insert a whole given the relatice details, into the correct side.
+
+        Args:
+            pos (tuple[float, float]): this is a tupple that contains the (x, y) coordinates of the object.
+            diameter (float): The diameter of the hole.
+            depth (float, optional): How deep to drill the hole, if not specified will drill the hole all the way through.
+            inner (bool, optional): This is specified that the hole is internal and will not be coppied onto other surface. Defaults to True.
+            external_only (bool, optional): This is specified that the hole will only be tranferred onto other surfaces and will not be drilled into main object. Defaults to False.
+        """
+        _depth = self._depth_check(depth)
+        _location_tupple = self._location_calc(pos=pos, sink=0)
+        self._parent.make_hole(
+            x=_location_tupple[0],
+            y=_location_tupple[1],
+            z=_location_tupple[2],
+            side=self.name,
+            diameter=diameter,
+            depth=_depth,
+            inner=inner,
+            external_only=external_only,
+        )
+
+    def box(
+        self,
+        pos: tuple[float, float],
+        length: float,
+        width: float,
+        depth: float = None,
+        sink: float = 0,
+        center: bool = False,
+    ):
+        """
+        This box will insert a rectangle shape cut out into the object.
+
+        Args:
+            pos (tuple[float, float]): The (x, y) coordinates of the box.
```

Comment:
Dont define the type in the docstring, mkdocs will error. It gets the type from the method signature.